### ISSUE TITLE
G function training for other blur kernel family

### ISSUE THREAD

**Issue by Laviness** [VinAIResearch/blur-kernel-space-exploring#4]
In Readme, the author said “To do image deblurring, data augmentation, and blur generation, you first need to train the blur encoding network (The F function in the paper). This is the only network that you need to train.“ However, after I retrained the F function using out-of-focus blur- sharp image pairs and used it in the image deblurring script "generic_deblur.py" to deblur out-of-focus images, the results are not yet satisfying. 

I trained with a batch_size of 13 and iteration of 600,000. The final loss is around 1e03, and the training datasets volume is around 3000 image pairs. 

I wonder if I need to retrain the G function as well, if I am using the network to deblur out-of-focus images. If needed, is it possible to provide a script that trains both G and F functions? 

Thank you!

**Comment by P0lyFish** [comment]
Thank you for your interest!
The functions F and G are trained simultaneously. So if you use the provided code to train F, G is trained too. Can you test the blur transferring code first to see if the functions F and G are trained correctly?
Also, to do image deblurring, it is required to carefully tune hyper-parameters, and we haven't tried the method with out-of-focus blur yet.
